integration-tests/src/test_loop/utils/transactions.rs

```diff
@@ -201,3 +206,92 @@ pub fn get_node_data<'a>(node_datas: &'a [TestData], account_id: &AccountId) ->
     }
     panic!("RPC client not found");
 }
+
+/// Run a transaction until completion and assert that the result is "success".
+/// Returns the transaction result.
+pub fn run_tx(
+    test_loop: &mut TestLoopV2,
+    tx: SignedTransaction,
+    node_datas: &[TestData],
+    maximum_duration: Duration,
+) -> Vec<u8> {
+    let tx_res = execute_tx(test_loop, tx, node_datas, maximum_duration).unwrap();
+    assert_matches!(tx_res.status, FinalExecutionStatus::SuccessValue(_));
+    match tx_res.status {
+        FinalExecutionStatus::SuccessValue(res) => res,
+        _ => unreachable!(),
+    }
+}
+
+/// Submit a transaction and wait for the execution result.
+/// For invalid transactions returns an error.
+/// For valid transactions returns the execution result (which could have an execution error inside, check it!).
+pub fn execute_tx(
+    test_loop: &mut TestLoopV2,
+    tx: SignedTransaction,
+    node_datas: &[TestData],
+    maximum_duration: Duration,
+) -> Result<FinalExecutionOutcomeView, InvalidTxError> {
+    // Last node is usually the rpc node
+    let rpc_node_id = node_datas.len().checked_sub(1).unwrap();
+
+    let tx_hash = tx.get_hash();
+
+    let process_result = Arc::new(Mutex::new(None));
+    let process_result_clone = process_result.clone();
+
+    node_datas[rpc_node_id].client_sender.clone().send(MessageWithCallback {
```

Comment:
Checking, do we need to clone client_sender?

Reply:
You're right, it works without the `.clone()`. I copied it from other code in `transactions.rs`.
Fixed, removed all such clones in the file.
It's interesting that clippy didn't pick this up :thinking: